### DISCUSSION TITLE
Lesson 2: why do we map outside the array?

### DISCUSSION THREAD

**Discussion by ZaynTawfik** [smartcontractkit/full-blockchain-solidity-course-py#1375] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/24631945/161891482-be6e9944-5555-43a2-96ad-d441f4cda15d.png)
In addPerson function, we first add a person using the push function to the People[] array we created. And then we map the name to its Favourite number. But that has got nothing to do with the array right? The array still only knows the index of the name and number that we just added. So where is this mapping being stored in the memory?

**Comment by Noveleader** [comment, marked as answer]
```People``` is a struct and with a ```uint256``` and a ```string``` and this struct is pushed into the array ```people```, hence creating a ```2D array```.
and mapping indicates its key as name ```string``` and value as ```uint256```

**Reply by cromewar** [reply]
@Noveleader is Completely right

**Reply by Noveleader** [reply]
Nice it helped
